### PR TITLE
[ML-DataFrame] improve error message for timeout case in stop

### PR DESCRIPTION
**Pull request by hendrikmuhs**
improve error message if stopping of transform times out.

related #45610

### CODE REVIEW

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/2

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/2

**Comment by hendrikmuhs** [comment]
The original issue I tried to address gets fixed with #46162, nevertheless the error handling improvements seem useful to me.

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportStopDataFrameTransformAction.java [hunk 1]
```diff
@@ -254,7 +254,52 @@ private void waitForDataFrameStopped(Set<String> persistentTaskIds,
 
                 listener.onFailure(new ElasticsearchStatusException(message, RestStatus.CONFLICT));
             },
-            listener::onFailure
+            e-> {
```
Comment (benwtrent) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportStopDataFrameTransformAction.java]:
```suggestion
            e -> {
```

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportStopDataFrameTransformAction.java [hunk 2]
```diff
@@ -254,7 +254,52 @@ private void waitForDataFrameStopped(Set<String> persistentTaskIds,
...
+                // waitForPersistentTasksCondition throws a IllegalStateException on timeout
+                if (e instanceof IllegalStateException && e.getMessage().startsWith("Timed out")) {
+                    PersistentTasksCustomMetaData persistentTasksCustomMetaData = clusterService.state().metaData()
+                        .custom(PersistentTasksCustomMetaData.TYPE);
+
+                    if (persistentTasksCustomMetaData == null) {
+                        listener.onResponse(new Response(Boolean.TRUE));
```
Comment (benwtrent) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportStopDataFrameTransformAction.java]:
```suggestion
                        listener.onResponse(new Response(Boolean.TRUE));
                        return;
```

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportStopDataFrameTransformAction.java [hunk 3]
```diff
@@ -254,7 +254,52 @@ private void waitForDataFrameStopped(Set<String> persistentTaskIds,
...
+
+                        if (stillRunningTasks.size() > 0) {
+                            message.append("Could not stop the transforms ");
+                            message.append(stillRunningTasks);
+                            message.append(" as they timed out.");
+                        }
+
+                        listener.onFailure(new ElasticsearchStatusException(message.toString(), RestStatus.REQUEST_TIMEOUT));
+                    }
+                }
```
Comment (benwtrent) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportStopDataFrameTransformAction.java]:
```suggestion
                    return;
                }
```
Or wrap the other `listener.onFailure(e);` in an else block. Seems to me there is a chance for the listener to be called twice.

**Review by benwtrent — APPROVED** [review]

**Pull request merged**